common/src/main/java/com/wynntils/wynn/handleditems/annotators/gui/IngredientPouchAnnotator.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wynn.handleditems.annotators.gui;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Managers;
+import com.wynntils.handlers.item.ItemAnnotation;
+import com.wynntils.handlers.item.ItemAnnotator;
+import com.wynntils.mc.utils.ItemUtils;
+import com.wynntils.utils.Pair;
+import com.wynntils.wynn.handleditems.items.gui.IngredientPouchItem;
+import com.wynntils.wynn.objects.profiles.ingredient.IngredientProfile;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+
+public final class IngredientPouchAnnotator implements ItemAnnotator {
+    private static final Pattern INGREDIENT_POUCH_PATTERN = Pattern.compile("^§6Ingredient Pouch$");
+    private static final Pattern INGREDIENT_LORE_LINE_PATTERN = Pattern.compile(
+            "^§f(\\d+) x (?:§r)?§7([^§]*)(?:§r)?(?:§[3567])? \\[(?:§r)?§([8bde])✫(?:§r)?(§8)?✫(?:§r)?(§8)?✫(?:§r)?§[3567]\\]$");
```

Comment:
Wow :upside_down_face:

Reply:
Yeah. It's ... deserves some kind of prize for "Best Example of Why Regular Expressions are Write-Only Code" :-)